HopsanGUI/Widgets/LibraryWidget.cpp

```diff
@@ -650,231 +515,199 @@ void LibraryWidget::handleItemClick(QTreeWidgetItem *item, int column)
-                            //! @todo fix the problem with save state
-                            auto spGenerator = createDefaultImportGenerator();
-                            bool compiledOK = spGenerator->compileComponentLibrary(libPath);
-                            if (!compiledOK)
-                            {
-                                gpMessageHandler->addErrorMessage("Library compiler failed");
-                            }
-
-                            // Now reload the library
-                            gpLibraryHandler->loadLibrary(libPath);
-                        }
-                    }
-                    gpModelHandler->restoreState();
-                }
-                // Handle check consistency
-                else if ((pReply == pCheckConsistenceAction) && !typeNames.isEmpty())
-                {
-                    ComponentLibraryEntry le = gpLibraryHandler->getEntry(typeNames.first());
-                    if (le.pLibrary)
-                    {
-                        auto spGenerator = createDefaultGenerator(false);
-                        bool checkOK = spGenerator->checkComponentLibrary(le.pLibrary->xmlFilePath);
-                        if (!checkOK) {
-                            gpMessageHandler->addWarningMessage(QString("The library '%1' has inconsistent component registration, this may cause exported models to fail.").arg(le.pLibrary->xmlFilePath));
-                        }
-                    }
-                }
+            else {
+                path = gpLibraryHandler->getModelObjectAppearancePtr(mItemToTypeNameMap.find(pFirstSubComponentItem).value())->getBasePath();
             }
-            else if(pReply == pOpenFolderAction)
-            {
-                QDesktopServices::openUrl(QUrl("file:///" + gpLibraryHandler->getModelObjectAppearancePtr(mItemToTypeNameMap.find(pFirstSubComponentItem).value())->getBasePath()));
+            QDesktopServices::openUrl(QUrl("file:///" + path));
+        }
+        else if(pReply == pEditXMLAction) {
+            if(!isComponentItem(item)) {
+                //Edit library XML file
+                SharedComponentLibraryPtrT pLibrary = mItemToLibraryMap[item];
+                if (pLibrary) {
+                    gpModelHandler->loadTextFile(pLibrary->xmlFilePath);
+                }
             }
-            else if(pReply == pEditXMLAction) {
+            else {
                 QFileInfo xmlFile = gpLibraryHandler->getModelObjectAppearancePtr(mItemToTypeNameMap.find(pFirstSubComponentItem).value())->getXMLFile();
                 gpModelHandler->loadTextFile(xmlFile.absoluteFilePath());
             }
-            else if(pReply == pEditCodeAction) {
+        }
+        else if(pReply == pEditCodeAction) {
+            if(!isComponentItem(item)) {
+                //Edit library source files
+                SharedComponentLibraryPtrT pLibrary = mItemToLibraryMap[item];
```

Comment:
not sure if this one should be editable (maybe read-only), since it should be auto generated

Reply:
I will make it const then for clarity.

Reply:
I meant, in HoLC I think it showed a meaasage in the editor that you should not edit the file since it will be overwriten everytime you change anything

Reply:
That's true. We don't have the read-only feature in HopsanGUI text editor yet. You may also want to edit this file, for example for temporarily commenting out a component that makes compilation fail. 

I guess we should not overwrite the entire file when we add/remove components, but instead append/remove the specific lines of code. This requires some clever code parsing though.